### PR TITLE
add 'playsinline' setting

### LINKED ISSUE #14: video always plays 'inline' on iPhone
course authors should have the ability to control this, as per the core media component

### PR DESCRIPTION
**Pull request by moloko**
replace example video with one that allows embedding on mobile

quick code tidy

resolves #14

### CODE REVIEW

File: properties.schema [hunk 1]
```diff
@@ -62,7 +62,7 @@
         "_controls": {
```
Comment (tomgreenfield) [inline properties.schema:65]: For the `_source` attribute above, suggest:
```json
"default": ""
```
and adding:
```json
"editorAttrs": {
  "placeholder": "//www.youtube.com/embed/jNQXAC9IVRw"
}
```

File: templates/youtube.hbs [hunk 2]
```diff
@@ -9,7 +9,19 @@
 
   <div class="component__widget youtube__widget">
 
-    <iframe width="640px" height="360px" style="width:100%;" src="{{_media._source}}?&autoplay={{#if _media._autoplay}}1{{else}}0{{/if}}&rel={{#if _media._showRelated}}1{{else}}0{{/if}}&loop={{#if _loop}}1{{else}}0{{/if}}&modestbranding={{#if _media._modestBranding}}1{{else}}0{{/if}}&color={{#if _media._progressColor}}{{_media._progressColor}}{{/if}}&controls={{#if _media._controls}}1{{else}}0{{/if}}&enablejsapi=1&playsinline=1&iv_load_policy={{#if _media._showAnnotations}}1{{else}}3{{/if}}" frameborder="0" {{#if _media._allowFullscreen}}allowfullscreen{{/if}}></iframe>
+    <iframe width="640px" height="360px" style="width:100%;"
```
Comment (tomgreenfield) [inline templates/youtube.hbs:13]: Suggest wrapping this in an `{{#if _media._source}}`.

At present when it's set to `""`/`undefined` (e.g. when you add the component in the authoring tool and don't edit the form), I get the course embedded within the component(!).

File: js/adapt-youtube.js [hunk 3]
```diff
@@ -44,19 +42,19 @@ define([
...
-      }
+      var aspectRatio = (this.model.get('_media')._aspectRatio ? parseFloat(this.model.get('_media')._aspectRatio) : 1.778);// default to 16:9 if not specified
+      if (isNaN(aspectRatio)) return;
+
+      $iframe.height(widgetWidth / aspectRatio);
     },
 
     postRender: function() {
-      //FOR HTML/HBS Paramenters: https://developers.google.com/youtube/player_parameters
+      // FOR HTML/HBS Paramenters: https://developers.google.com/youtube/player_parameters
```
Comment (tomgreenfield) [inline js/adapt-youtube.js]: Typo: `Paramenters`.

Comment (moloko) [reply js/adapt-youtube.js]: TBH not sure that comment is particularly useful in that location... might remove

File: templates/youtube.hbs (hunk 2 above)
Comment (moloko) [reply templates/youtube.hbs:13]: This is so cool I'm tempted to ignore your suggestion and leave it as-is 

image

Comment (tomgreenfield) [reply templates/youtube.hbs:13]: Slight problem of wrapping the entire `<iframe>` is that the code will error out before `_isReady` is set (infinite loading).

Could instead wrap only the `src` attribute, or do some defensive coding in the JS (might be more time-consuming for you).

Comment (moloko) [reply templates/youtube.hbs:13]: cheers for the heads-up, I'll have a a think about how best to handle this

Comment (moloko) [reply templates/youtube.hbs:13]: How about something like this?
image

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

File: js/adapt-youtube.js [hunk 4]
```diff
@@ -44,19 +42,18 @@ define([
 
       $iframe.width(widgetWidth);
 
-      var aspectRatio = (this.model.get('_media')._aspectRatio ? parseFloat(this.model.get('_media')._aspectRatio) : 1.778);//default to 16:9 if not specified
-      if (!isNaN(aspectRatio)) {
-        $iframe.height(widgetWidth / aspectRatio);
-      }
+      var aspectRatio = (this.model.get('_media')._aspectRatio ? parseFloat(this.model.get('_media')._aspectRatio) : 1.778);// default to 16:9 if not specified
+      if (isNaN(aspectRatio)) return;
+
```
Comment (eleanor-heath) [inline js/adapt-youtube.js:47]: is this empty line needed? Also I would place this comment `// default to 16:9 if not specified` above where the aspectRatio variable is initialised.

File: js/adapt-youtube.js [hunk 5]
```diff
@@ -102,9 +99,9 @@ define([
     },
 
     onPlayerReady: function() {
-      if (this.model.get('_media')._playbackQuality) {
-        this.player.setPlaybackQuality(this.model.get('_media')._playbackQuality);
-      }
+      if (!this.model.get('_media')._playbackQuality) return;
+
```
Comment (eleanor-heath) [inline js/adapt-youtube.js:103]: is this space needed?

File: js/adapt-youtube.js [hunk 6]
```diff
@@ -116,38 +113,36 @@ define([
...
+          break;
       }
     },
 
     onSkipToTranscript: function() {
       // need slight delay before focussing button to make it work when JAWS is running
       // see https://github.com/adaptlearning/adapt_framework/issues/2427
       _.delay(function() {
-        this.$('.youtube__transcript-btn').a11y_focus();
+        Adapt.a11y.focusFirst(this.$('.youtube__transcript-btn'));
```
Comment (eleanor-heath) [inline js/adapt-youtube.js:145]: you could do `Adapt.a11y.focusFirst('.youtube__transcript-btn');` I believe.

**Review by eleanor-heath — APPROVED** [review]

**Comment by oliverfoster** [comment]
closed in favour of https://github.com/adaptlearning/adapt-youtube/pull/17

**Pull request closed without merging**